### ISSUE TITLE
stdlib docs generation refactor

### ISSUE THREAD

**Issue by vlopes11** [maticnetwork/miden#464]
Currently we have are using different pipelines to build the id |-> ast mapping and the docs from its asm source

We should refactor this structure to have a single pipeline so we don't run the risk of having one outdated in relation to the other

_Originally posted by @vlopes11 in https://github.com/maticnetwork/miden/pull/461#discussion_r1012764311_

**Comment by bobbinth** [comment]
The idea is that this will be available at the AST level. For example, `ProcedureAst` will look something like this:
```Rust
pub struct ProcedureAst {
    pub name: String,
    pub docs: Option<String>,
    pub num_locals: u32,
    pub body: Vec<Node>,
    pub is_export: bool,
    pub index: u32,
}
```

Where `docs` would contain doc comments associated with the procedures.

**Comment by bobbinth** [comment]
Closed by #494
